### PR TITLE
enable service-dir lwip tests

### PR DESCRIPTION
**Pull request by jenniexie**
Hi Jay/RS:
    Based on last week's meeting, I've made changes to service-dir lwip test and tested at my local for both lwip and socket case, test passed in both cases in my local test.

Please help to review the code.

BTW: This will need another small PR from Happy, which will remove IP assignment if device is a tap device.
https://github.com/openweave/happy/pull/35

### CODE REVIEW

File: src/test-apps/happy/test-templates/WeaveServiceDir.py [hunk 1]
```diff
@@ -133,13 +133,12 @@ def __pre_check(self):
         self.client_ip = self.getNodeWeaveIPAddress(self.client)
 
         # Check if all unknowns were found
-
-        if self.service_ip == None:
+        if self.service_ip == None and self.use_lwip == False:
```
Comment (robszewczyk) [inline src/test-apps/happy/test-templates/WeaveServiceDir.py]: Style: `and not self.use_lwip`

File: src/test-apps/happy/test-templates/WeaveServiceDir.py [hunk 2]
```diff
@@ -133,13 +133,12 @@ def __pre_check(self):
...
             emsg = "Could not find IP address of the service."
             self.logger.error("[localhost] WeaveServiceDir: %s" % (emsg))
             self.exit()
 
-        if self.client_ip == None:
+        if self.client_ip == None and self.use_lwip == False:
```
Comment (robszewczyk) [inline src/test-apps/happy/test-templates/WeaveServiceDir.py]: Style: `and not self.use_lwip`

File: src/test-apps/happy/topologies/standalone/thread_wifi_on_tap_ap_service.sh [hunk 3]
```diff
@@ -22,7 +22,7 @@
 # - one thread-wifi border_gateway device BorderRouter
 # - one access-point router onhub
 # - one service node instance cloud
-
+export PATH=$PATH:$TRAVIS_BUILD_DIR/src/test-apps/happy/bin
```
Comment (robszewczyk) [inline src/test-apps/happy/topologies/standalone/thread_wifi_on_tap_ap_service.sh]: `$TRAVIS_BUILD_DIR` now makes this specific to the Travis CI integration. please use something more general

File: src/test-apps/happy/topologies/standalone/thread_wifi_ap_service.sh [hunk 4]
```diff
@@ -23,6 +23,7 @@
 # - one access-point router onhub
 # - one service node instance cloud
 
+export PATH=$PATH:$TRAVIS_BUILD_DIR/src/test-apps/happy/bin
```
Comment (robszewczyk) [inline src/test-apps/happy/topologies/standalone/thread_wifi_ap_service.sh]: `$TRAVIS_BUILD_DIR`  makes this specific to the Travis CI integration. please use something more general

File: src/test-apps/happy/test-templates/WeaveServiceDir.py (hunk 1 above)
Comment (jenniexie) [reply src/test-apps/happy/test-templates/WeaveServiceDir.py]: done

File: src/test-apps/happy/test-templates/WeaveServiceDir.py (hunk 2 above)
Comment (jenniexie) [reply src/test-apps/happy/test-templates/WeaveServiceDir.py]: done

File: src/test-apps/happy/topologies/standalone/thread_wifi_on_tap_ap_service.sh (hunk 3 above)
Comment (jenniexie) [reply src/test-apps/happy/topologies/standalone/thread_wifi_on_tap_ap_service.sh]: done

File: src/test-apps/happy/topologies/standalone/thread_wifi_ap_service.sh (hunk 4 above)
Comment (jenniexie) [reply src/test-apps/happy/topologies/standalone/thread_wifi_ap_service.sh]: done

**Comment by jenniexie** [comment]
Thanks RS for your feedback!
I have addressed all comments, please help to review again.

**Comment by jenniexie** [comment]
> Overall, when I last found myself in the depths of all of this code, I found the lack of comments incredibly lacking.
> 
> In the interest of a well-managed and -administered open source project, I'd advocate "leaving things in a better state than we found them" in this regard. Consequently, I'd advocate augmenting the code changes here with qualitative comments detailing "why" things are being done.
> 
> Hopefully, this will make it easier for subsequent visitors here to understand and more easily change or address the implementation in the future.

sure, will do

**Comment by jenniexie** [comment]
Base on offline discussion with @gerickson yesterday, I will work on issue #307 in a separate PR https://github.com/openweave/openweave-core/issues/307

Thanks @gerickson for your feedback!

**Pull request merged**